bin/ch-run.c

```diff
@@ -184,6 +187,36 @@ void env_delta_append(struct env_delta **ds, enum env_action act, char *arg)
    (*ds)[i].arg = arg;
 }
 
+/* Parse and set input from --set-env */
+void env_parse(char *name, char *new_value, bool expand)
+{
+   char *token;
+   char *new_env = "";
+   const char s[1] = ":";
```

Comment:
This variable is used twice and gives you room to change the separator, but I think that amount of indirection outweighs the additional cognitive load it adds. I'd just use literals below with `strtok()`.

Also, it's an unterminated string b/c the length is 1.